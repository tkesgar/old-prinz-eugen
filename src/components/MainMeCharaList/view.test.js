import { mount } from '@vue/test-utils'
import localVue from '../../utils/test/local-vue'
import MainMeCharaListView from './view'

describe('MainMeCharaListView', () => {
  describe('with empty chara list (default)', () => {
    const wrapper = mount(MainMeCharaListView, { localVue })

    test('should mount successfully', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })

    test('should match snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })

    test('should show empty message', () => {
      expect(wrapper.find('[data-test-key="empty-message"]').exists()).toBe(true)
    })
  })

  describe('with chara list', () => {
    const charas = [
      { id: 1, name: 'admiral_hipper' },
      { id: 2, name: 'blucher' },
      { id: 3, name: 'prinz_eugen' },
      { id: 4, name: 'seydlitz' },
      { id: 5, name: 'lutzow' }
    ]

    const wrapper = mount(MainMeCharaListView, {
      localVue,
      propsData: { charas }
    })

    test('should mount successfully', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })

    test('should match snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })

    test('should shows the correct amount of chara entry', () => {
      expect(wrapper.findAll('[data-test-key="chara"]').length).toBe(charas.length)
    })
  })
})
