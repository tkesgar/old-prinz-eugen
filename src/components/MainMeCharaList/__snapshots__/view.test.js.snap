// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMeCharaListView with chara list should match snapshot 1`] = `
<div>
  <div
    data-test-key="chara"
  >
    <a
      class=""
      href="/chara/1"
      target="_self"
    >
      admiral_hipper
    </a>
  </div>
  <div
    data-test-key="chara"
  >
    <a
      class=""
      href="/chara/2"
      target="_self"
    >
      blucher
    </a>
  </div>
  <div
    data-test-key="chara"
  >
    <a
      class=""
      href="/chara/3"
      target="_self"
    >
      prinz_eugen
    </a>
  </div>
  <div
    data-test-key="chara"
  >
    <a
      class=""
      href="/chara/4"
      target="_self"
    >
      seydlitz
    </a>
  </div>
  <div
    data-test-key="chara"
  >
    <a
      class=""
      href="/chara/5"
      target="_self"
    >
      lutzow
    </a>
  </div>
</div>
`;

exports[`MainMeCharaListView with empty chara list (default) should match snapshot 1`] = `
<div>
  <div
    class="text-center"
    data-test-key="empty-message"
  >
    <p>
      Belum ada karakter.
    </p>
     
    <button
      class="btn btn-primary"
      type="button"
    >
      Buat karakter baru
    </button>
  </div>
</div>
`;
